provider/value_cannot_be_changed_modifier.go

```diff
@@ -0,0 +1,90 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/openshift-online/ocm-sdk-go/logging"
+)
+
+type valueCannotBeChangedModifier struct {
+	logger logging.Logger
+}
+
+func ValueCannotBeChangedModifier(logger logging.Logger) tfsdk.AttributePlanModifier {
+	return valueCannotBeChangedModifier{
+		logger: logger,
+	}
+}
+func (m valueCannotBeChangedModifier) Description(ctx context.Context) string {
+	return "Value does not change after creation."
+}
+
+func (m valueCannotBeChangedModifier) MarkdownDescription(ctx context.Context) string {
+	return m.Description(ctx)
+}
+
+func (m valueCannotBeChangedModifier) Modify(ctx context.Context, req tfsdk.ModifyAttributePlanRequest, resp *tfsdk.ModifyAttributePlanResponse) {
+
+	if req.AttributeConfig == nil || req.AttributeState == nil || req.AttributePlan == nil {
+		// shouldn't happen, but let's not panic if it does
+		return
+	}
+
+	if req.State.Raw.IsNull() {
+		// if we're creating the resource, no need to delete and
+		// recreate it
+		return
+	}
+
+	if req.Plan.Raw.IsNull() {
+		// if we're deleting the resource, no need to delete and
+		// recreate it
+		return
+	}
+
+	attrSchema, err := req.State.Schema.AttributeAtPath(req.AttributePath)
+	if err != nil && !errors.Is(err, errors.New("path leads to block, not an attribute")) {
+		resp.Diagnostics.AddAttributeError(req.AttributePath,
+			"Error finding attribute schema",
+			fmt.Sprintf("An unexpected error was encountered retrieving the schema for this attribute. This is always a bug in the provider.\n\nError: %s", err),
+		)
+		return
+	}
+
+	configRaw, err := req.AttributeConfig.ToTerraformValue(ctx)
+	if err != nil {
+		resp.Diagnostics.AddAttributeError(req.AttributePath,
+			"Error converting config value",
+			fmt.Sprintf("An unexpected error was encountered converting a %s to its equivalent Terraform representation. This is always a bug in the provider.\n\nError: %s", req.AttributeConfig.Type(ctx), err),
+		)
+		return
+	}
+
+	if configRaw == nil && attrSchema.Computed {
+		// if the config is null and the attribute is computed, this
+		// could be an out-of-band change, don't require blocking
+		return
+	}
+
+	v, err := req.AttributeState.ToTerraformValue(ctx)
+	if err == nil {
+		m.logger.Debug(ctx, fmt.Sprintf("attribute state is %v", v))
+	}
+	v, err = req.AttributePlan.ToTerraformValue(ctx)
+	if err == nil {
+		m.logger.Debug(ctx, fmt.Sprintf("attribute plan is %v", v))
+	}
```

Comment:
Its not clear to me what those lines used for - v is never used
If it created only for logs, consider to remove it

Reply:
I removed those logger.debug lines